src/rust/engine/src/rule_graph.rs

```diff
@@ -332,43 +344,41 @@ impl <'a> GraphMaker<'a> {
                                      entry.subject_type(),
                                      format!("no matches for {:?} when resolving {:?}", entry.subject_type(), initial_selector)); // might be better as {} with display derived
                   was_unfulfillable = true;
-                  continue
+                  continue;
                 }
-                // TODO port the rest of this after adding field types to the selectors
-
-                /*
-                      initial_selector = selector.input_product_selector
-          initial_rules_or_literals = _find_rhs_for_select(entry.subject_type, initial_selector)
-          if not initial_rules_or_literals:
-            mark_unfulfillable(entry,
-                               entry.subject_type,
-                               'no matches for {} when resolving {}'
-                               .format(initial_selector, selector))
-            was_unfulfillable = True
-            continue
-
-          rules_for_dependencies = []
-          for field_type in selector.field_types:
-            rules_for_field_subjects = _find_rhs_for_select(field_type,
-                                                            selector.projected_product_selector)
-            rules_for_dependencies.extend(rules_for_field_subjects)
-
-          if not rules_for_dependencies:
-            mark_unfulfillable(entry,
-                               selector.field_types,
-                               'no matches for {} when resolving {}'
-                               .format(selector.projected_product_selector, selector))
-            was_unfulfillable = True
-            continue
-
-          add_rules_to_graph(entry,
-                             (selector, selector.input_product_selector),
-                             initial_rules_or_literals)
-          add_rules_to_graph(entry,
-                             (selector, selector.projected_product_selector),
-                             tuple(rules_for_dependencies))
-                             */
+                let mut rules_for_dependencies = vec![];
+                for field_type in select.field_types.iter() {
```

Comment:
In many cases, calling `iter` should not be necessary? I think anything with `Into<Iterator>` will work in this position (and an `Iterator` works because it has an `Into<Iterator>` impl.)